src/main/java/net/sf/jabref/logic/util/io/FileHistory.java

```diff
@@ -49,9 +49,9 @@ public int size() {
 
     public void newFile(String filename) {
         history.remove(filename);
-        history.addFirst(filename);
+        ((LinkedList<String>) history).addFirst(filename);
         while (history.size() > HISTORY_SIZE) {
-            history.removeLast();
+            ((LinkedList<String>) history).removeLast();
```

Comment:
In that case, I would leave it at LinkedList as we make explicit use of this data structure.